TST: test_two_sum_size fails in s390x with f1=-7.303128360378417e+307 and f2=1.7976931348623157e+308

Example log: https://github.com/astropy/astropy/actions/runs/4593442983/jobs/8111372457

```
______________________________ test_two_sum_size _______________________________
  
      @given(
  >       floats(allow_nan=False, allow_infinity=False),
          floats(allow_nan=False, allow_infinity=False),
      )
  
  astropy/time/tests/test_precision.py:342: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
  f1 = -7.303128360378417e+307, f2 = 1.7976931348623157e+308
  
      @given(
          floats(allow_nan=False, allow_infinity=False),
          floats(allow_nan=False, allow_infinity=False),
      )
      @example(f1=8.988465674311579e307, f2=8.98846567431158e307)
      @example(f1=8.988465674311579e307, f2=-8.98846567431158e307)
      @example(f1=-8.988465674311579e307, f2=-8.98846567431158e307)
      def test_two_sum_size(f1, f2):
          r1, r2 = two_sum(f1, f2)
  >       assert (
              abs(r1) > abs(r2) / np.finfo(float).eps
              or r1 == r2 == 0
              or not np.isfinite(f1 + f2)
          )
  E       AssertionError: assert (1.0673802988244741e+308 > (nan / 2.220446049250313e-16) or 1.0673802988244741e+308 == nan or not True)
  E        +  where 1.0673802988244741e+308 = abs(1.0673802988244741e+308)
  E        +  and   nan = abs(nan)
  E        +  and   2.220446049250313e-16 = finfo(resolution=1e-15, min=-1.7976931348623157e+308, max=1.7976931348623157e+308, dtype=float64).eps
  E        +    where finfo(resolution=1e-15, min=-1.7976931348623157e+308, max=1.7976931348623157e+308, dtype=float64) = <class 'numpy.finfo'>(float)
  E        +      where <class 'numpy.finfo'> = np.finfo
  E        +  and   True = <ufunc 'isfinite'>((-7.303128360378417e+307 + 1.7976931348623157e+308))
  E        +    where <ufunc 'isfinite'> = np.isfinite
  E       Falsifying example: test_two_sum_size(
  E           f1=-7.303128360378417e+307,
  E           f2=1.7976931348623157e+308,
  E       )
  E       
  E       You can reproduce this example by temporarily adding @reproduce_failure('6.70.2', b'AAAAAb/qAAAAAAEDAAAAv/////////8=') as a decorator on your test case
  
  astropy/time/tests/test_precision.py:350: AssertionError
  =========================== short test summary info ============================
  FAILED astropy/time/tests/test_precision.py::test_two_sum_size - AssertionError: assert (1.0673802988244741e+308 > (nan / 2.220446049250313e-16) or 1.0673802988244741e+308 == nan or not True)
   +  where 1.0673802988244741e+308 = abs(1.0673802988244741e+308)
   +  and   nan = abs(nan)
   +  and   2.220446049250313e-16 = finfo(resolution=1e-15, min=-1.79769313486[2315](https://github.com/astropy/astropy/actions/runs/4593442983/jobs/8111372457#step:3:2316)7e+308, max=1.7976931348623157e+308, dtype=float64).eps
   +    where finfo(resolution=1e-15, min=-1.7976931348623157e+308, max=1.7976931348623157e+308, dtype=float64) = <class 'numpy.finfo'>(float)
   +      where <class 'numpy.finfo'> = np.finfo
   +  and   True = <ufunc 'isfinite'>((-7.303128360378417e+307 + 1.7976931348623157e+308))
   +    where <ufunc 'isfinite'> = np.isfinite
  Falsifying example: test_two_sum_size(
      f1=-7.303128360378417e+307,
      f2=1.7976931348623157e+308,
  )
  
  You can reproduce this example by temporarily adding @reproduce_failure('6.70.2', b'AAAAAb/qAAAAAAEDAAAAv/////////8=') as a decorator on your test case
```

--- a/astropy/time/tests/test_precision.py
+++ b/astropy/time/tests/test_precision.py
@@ -345,12 +345,13 @@ def test_two_sum_symmetric(f1, f2):
 @example(f1=8.988465674311579e307, f2=8.98846567431158e307)
 @example(f1=8.988465674311579e307, f2=-8.98846567431158e307)
 @example(f1=-8.988465674311579e307, f2=-8.98846567431158e307)
+@example(f1=-7.303128360378417e307, f2=1.7976931348623157e308)
 def test_two_sum_size(f1, f2):
     r1, r2 = two_sum(f1, f2)
     assert (
         abs(r1) > abs(r2) / np.finfo(float).eps
         or r1 == r2 == 0
-        or not np.isfinite(f1 + f2)
+        or not np.isfinite(abs(f1) + abs(f2))
     )
 
 

